### ISSUE TITLE
Add Unfold?

### ISSUE THREAD

**Issue by bender2k14** [louthy/language-ext#697]
F# has a function called [`Seq.unfold`](https://msdn.microsoft.com/visualfsharpdocs/conceptual/seq.unfold%5b%27state%2c%27t%5d-function-%5bfsharp%5d).  I just encountered a problem that was perfectly solved by `Unfold`, and I used this function for the first time.  I couldn't find it in Language Ext, so I wrote my own.  It needs to be iterative to avoid overflowing the stack.

```C#
public static IEnumerable<A> Unfold<A, S>(
  Func<S, Option<(A, S)>> f,
  S seed
) {
  var mp = f(seed);
  while (mp.IsSome) {
    var (a, s) = mp.IfNoneUnsafe(null);
    yield return a;
    mp = f(s);
  }
}
```

If anyone else would find this useful, then maybe we could add it to Language Ext.

**Comment by louthy** [comment]
Already exists 👍 

https://github.com/louthy/language-ext/blob/master/LanguageExt.Core/DataTypes/List/Lst.Module.cs#L1067

**Comment by bender2k14** [comment]
Oh, great.  Thanks :)  I must have had a typo in my search query.  I am finding it now when searching.

What about adding overloads that use `ValueTuple`?

**Comment by bender2k14** [comment]
> What about adding overloads that use `ValueTuple`?

And released in [3.3.49](https://github.com/louthy/language-ext/releases/tag/v3.3.49); that was fast!

Thanks :D
